pkg/controller/ippool/controller.go

```diff
@@ -152,3 +157,63 @@ func correctAllocatedHistory(pool *lbv1.IPPool) (map[string]string, error) {
 
 	return allocatedHistory, nil
 }
+
+// Due to previous version bug, the LB may be removed but the related IP still exists on pool allocation record
+// This feature gives a way to remove the IP allocation record manually
+func (h *Handler) OnChangeToReleaseAnIP(_ string, ipPool *lbv1.IPPool) (*lbv1.IPPool, error) {
+	if ipPool == nil || ipPool.DeletionTimestamp != nil || ipPool.Annotations == nil {
+		return ipPool, nil
+	}
+
+	ipStr := ipPool.Annotations[utils.AnnotationKeyManuallyReleaseIP]
+	if ipStr == "" {
+		return ipPool, nil
+	}
+
+	// check if it is a valid format, e.g.
+	// loadbalancer.harvesterhci.io/manuallyReleaseIP: "192.168.5.12: default/cluster1-lb-3"
+	ip, namespace, name, err := utils.SplitIPAllocatedString(ipStr)
+	if err != nil {
+		logrus.Infof("IP Pool %s has a manual IP release request %s, it is not valid, skip", ipPool.Name, ipStr)
+		return h.removeAnnotationKeyManuallyReleaseIP(ipPool)
+	}
+
+	// check it is a real allocation record
+	lbStr := ipPool.Status.Allocated[ip]
+	if lbStr == "" || lbStr != fmt.Sprintf("%s/%s", namespace, name) {
+		logrus.Infof("IP Pool %s has a manual IP release request %s, it has been released, skip", ipPool.Name, ipStr)
+		return h.removeAnnotationKeyManuallyReleaseIP(ipPool)
+	}
+
+	// check if the lb is existing
+	_, err = h.lbCache.Get(namespace, name)
+	if err == nil {
+		logrus.Infof("IP Pool %s has a manual IP release request %s, the lb %s/%s is still existing, skip", ipPool.Name, ipStr, namespace, name)
+		return h.removeAnnotationKeyManuallyReleaseIP(ipPool)
+	}
+
+	// return for retry
+	if !apierrors.IsNotFound(err) {
+		return ipPool, err
+	}
+
+	a := h.allocatorMap.Get(ipPool.Name)
+	if a == nil {
+		return ipPool, fmt.Errorf("IP Pool %s has a manual IP release request %s, fail to get allocator", ipPool.Name, ipStr)
```

Comment:
If there is an IPPool, but there is no internal allocator, can we just create a new allocator for it and leave some log? If we return error directly, the controller may keep retrying.